### ISSUE TITLE
Conditionally display action buttons

### ISSUE THREAD

**Issue by clickanidea** [Power-Components/livewire-powergrid#136]
Is it possible to display an action button based on a condition ?
Thanks

**Comment by luanfreitasdev** [comment]
Hello @clickanidea , Not yet

Would it be something like that? In that case it would display the edit button when it is active.
```php
return [
       Button::add('edit')
            ->caption(__('Edit'))
            ->can($can)
            ->when(function (Dish $dish) {
                   return $dish->status === Dish::ACTIVE;
            })
            ->class($btnEditClass)
            ->openModal('edit-dish', [ 'dishId' => 'id']),
]
```

**Comment by clickanidea** [comment]
See below

<img width="1005" alt="CleanShot 2021-11-20 at 20 49 15@2x" src="https://user-images.githubusercontent.com/70272902/142737800-68250271-b007-49b0-9d6e-217ee6328296.png">

```php
public function actions(): array
    {
        return [
            Button::add('view')
                ->caption('View')
                ->class('bg-teal-500 cursor-pointer text-white px-3 py-2 m-1 rounded text-sm')
                ->when(function (Listing $listing) {
                    return $listing->is_approved === Listing::ACTIVE;
                })
                ->emit('pubblicaInserzione', ['key' => 'id'])
        ];
    }
```

**Comment by luanfreitasdev** [comment]
Hi, this feature is very useful but it will stay for the near future.

For now I will close the issue, I'll let you know if I make an advance

**Comment by dansysanalyst** [comment]
@clickanidea, this feature was made available in version 2.2.0.

Read more here:
https://livewire-powergrid.com/#/table/action-rules?id=action-rules

Thank you for the suggestion!

**Comment by long-blade** [comment]
@dansysanalyst and how we handle this in 4.x version ?
